### DISCUSSION TITLE
Spread_Rate Modelling Fire

### DISCUSSION THREAD

**Discussion by UnknownFdsUser** [firemodels/fds#13736] [Q&A]
Good evening, 

Wanting to use al Spread_Rate function in FDS I find a publication where the velocity in m/s is calculated as below: 

𝑆𝑃𝑅𝐸𝐴𝐷_𝑅𝐴𝑇𝐸=sqrt((𝛼/𝜋)∙(1/𝐻𝑅𝑅𝑃𝑈𝐴)) where α = Alpha-t2 growth rate (kW/s2)

My Fire has a peak  of 5,000 kW distributed over a square area of 1.80 m x 1.80 m. 

HRRPUA = 5000 kW /3.24sqm =1543 kW/sqm

𝛼= 1,000 kW /150 s^2 = 0.04444 kW/s^2

Mesh 0.20 m x 0.20 m x 0.20 m

Applying the above formula I will have a Spread_Rate velocity of 0.0030 m/s. 

By entering this value into FDS, after 150 s I do not have 1,000 kW. I cannot figure out where the error is...    
[S_R.zip](https://github.com/user-attachments/files/17692507/S_R.zip)

**Comment by drjfloyd** [comment]
SPREAD_RATE defines the time when grid cells ignite and start burning. Once those cells start burning, they then follow the HRRPUA and RAMP for the SURF. Each cell follows the RAMP based on the time it ignites. After 150 s, the first cell to ignite will have an HRRPUA of 1543 * 0.2 (The RAMP value at 150 s).  The cells at a distance of 0.45 m (150 * 0.003) which and an HRRPUA of 1543 * 0 (The RAMP value at 150 s).

**Reply by UnknownFdsUser** [reply]
Thank you drjfloyd

I created the ramp that if it was a t-squared fire: 
1543 kW/sq m * 1.80 m x 1.80 m * 0.20 (ramp value at 150 s)= 999.864 kW about 1,000 kW and thus Fast growth rate (1000 kW @ 150 s). 
What is the logic then to create a ramp with fast growth that reaches a peak power of 5 MW with Spread_Rate?

**Reply by drjfloyd** [reply]
-Just use SPREAD_RATE without a RAMP. A fire with a fixed HRRPUA that has a radius growining linearly with time will be a t^2 fire. This approach; however, will not give a smooth curve but rather a stepwise curve the increases each time a new grid cell ignites.
-Don't use SPREAD_RATE and just have a RAMP.  Your RAMP is a t^2 RAMP.
-Look at a recently created verification case that shows how to use CTRL logic to combine SPREAD_RATE with a t^2 RAMP. https://github.com/firemodels/fds/blob/master/Verification/Controls/spreading_fire.fds

**Reply by UnknownFdsUser** [reply]
Thank you very much, I have tried it and everything works as it should. I will study the attached fds file[(https://github.com/firemodels/fds/blob/master/Verification/Controls/spreading_fire.fds](https://github.com/firemodels/fds/blob/master/Verification/Controls/spreading_fire.fds)). 
Thank you as always for your time. 
Have a great day and enjoy your work!
